In [1]:
import pandas as pd

In [2]:
from pathlib import Path

file_path = Path(r"C:/Users/emree/sql-basic-practice/sgk_labor_market_excel_dashboard.xlsm")

file_path.exists()

True

In [3]:
service_production_sheet = "25-26-serviceproductionbysec"

df_service_production = pd.read_excel(
    file_path,
    sheet_name=service_production_sheet
)

df_service_production.head()

,Dönem,Ekonomik Faaliyet,Korunacak/Artacak,Azalacak,Fikrim Yok,Tbl_Anahtar
0,2026Q1,Eğitim,0.784,0.073,0.143,2026Q1|Eğitim
1,2026Q1,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.761,0.090,0.149,2026Q1|Konaklama ve Yiyecek Hizmeti Faaliyetleri
2,2026Q1,Gayrimenkul Faaliyetleri,0.756,0.077,0.167,2026Q1|Gayrimenkul Faaliyetleri
3,2026Q1,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.750,0.062,0.188,"2026Q1|Mesleki, Bilimsel ve Teknik Faaliyetler"
4,2026Q1,"Elektrik, Gaz, Buhar ve İklimlendirme Üretimi ...",0.744,0.043,0.213,"2026Q1|Elektrik, Gaz, Buhar ve İklimlendirme Ü..."


In [4]:
df_service_production.shape

(85, 6)

In [5]:
df_service_production.columns

Index(['Dönem', 'Ekonomik Faaliyet', 'Korunacak/Artacak', 'Azalacak',
       'Fikrim Yok', 'Tbl_Anahtar'],
      dtype='object')

In [6]:
df_service_production.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85 entries, 0 to 84
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Dönem              85 non-null     object 
 1   Ekonomik Faaliyet  85 non-null     object 
 2   Korunacak/Artacak  85 non-null     float64
 3   Azalacak           85 non-null     float64
 4   Fikrim Yok         85 non-null     float64
 5   Tbl_Anahtar        85 non-null     object 
dtypes: float64(3), object(3)
memory usage: 4.1+ KB


# Service Production by Sector Analysis

This notebook analyzes service/production expectations by economic activity and period.  
The goal is to compare sectors by maintain/increase expectations, decrease expectations, uncertainty levels, and overall service/production outlook.

In [8]:
df_service_production.columns = df_service_production.columns.str.strip()

df_service_production = df_service_production.rename (columns={
    "Dönem" : "period",
    "Ekonomik Faaliyet" : "economic_activity",
    "Korunacak/Artacak" : "maintain_or_increase_rate",
    "Azalacak" : "decrease_rate",
    "Fikrim Yok" : "no_opinion_rate",
    "Tbl_Anahtar" : "table_key"
})

df_service_production.head()

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate,table_key
0,2026Q1,Eğitim,0.784,0.073,0.143,2026Q1|Eğitim
1,2026Q1,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.761,0.090,0.149,2026Q1|Konaklama ve Yiyecek Hizmeti Faaliyetleri
2,2026Q1,Gayrimenkul Faaliyetleri,0.756,0.077,0.167,2026Q1|Gayrimenkul Faaliyetleri
3,2026Q1,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.750,0.062,0.188,"2026Q1|Mesleki, Bilimsel ve Teknik Faaliyetler"
4,2026Q1,"Elektrik, Gaz, Buhar ve İklimlendirme Üretimi ...",0.744,0.043,0.213,"2026Q1|Elektrik, Gaz, Buhar ve İklimlendirme Ü..."


## Data Quality Check

Before starting the analysis, missing values and duplicate rows are checked to understand the reliability of the dataset.

In [9]:
df_service_production.isnull().sum()

period                       0
economic_activity            0
maintain_or_increase_rate    0
decrease_rate                0
no_opinion_rate              0
table_key                    0
dtype: int64

In [11]:
df_service_production.duplicated().sum()

np.int64(0)

## Analysis 1: Period-Level Service/Production Expectation Summary

This analysis summarizes average service/production expectations by period.  
It helps compare whether expectations are becoming more positive or negative over time.

In [13]:
service_production_period_summary = df_service_production.groupby("period").agg(
    average_maintain_or_increase_rate = ("maintain_or_increase_rate","mean"),
    average_decrease_rate = ("decrease_rate", "mean"),
    average_no_opinion_rate = ("no_opinion_rate", "mean")
).reset_index()

service_production_period_summary

,period,average_maintain_or_increase_rate,average_decrease_rate,average_no_opinion_rate
0,2025Q1,0.700000,0.093235,0.206765
1,2025Q2,0.721941,0.072000,0.206118
2,2025Q3,0.725059,0.080882,0.194118
3,2025Q4,0.726706,0.082824,0.190412
4,2026Q1,0.727588,0.083412,0.188824


## Analysis 2: Sectors with the Highest Maintain or Increase Expectation

This analysis identifies the sectors with the highest expectation that service/production activity will be maintained or increased.

In [15]:
top_service_production_increase = df_service_production[
    ["period","economic_activity", "maintain_or_increase_rate", "decrease_rate","no_opinion_rate"]
].sort_values(by="maintain_or_increase_rate", ascending = False)

top_service_production_increase.head(10)  

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate
34,2025Q3,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.806,0.041,0.153
17,2025Q4,Bilgi ve İletişim,0.802,0.042,0.156
35,2025Q3,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.801,0.039,0.159
68,2025Q1,"Elektrik, Gaz, Buhar ve İklimlendirme Üretimi ...",0.797,0.037,0.167
18,2025Q4,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.792,0.050,0.157
51,2025Q2,Bilgi ve İletişim,0.790,0.047,0.163
69,2025Q1,Finans ve Sigorta Faaliyetleri,0.787,0.051,0.162
19,2025Q4,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.786,0.038,0.175
0,2026Q1,Eğitim,0.784,0.073,0.143
20,2025Q4,Eğitim,0.781,0.070,0.149


## Analysis 3: Sectors with the Highest Decrease Expectation

This analysis identifies the sectors where service/production decrease expectation is the highest.

In [16]:
top_service_production_decrease = df_service_production[
    ["period", "economic_activity", "maintain_or_increase_rate", "decrease_rate", "no_opinion_rate"]
].sort_values(by="decrease_rate", ascending=False)

top_service_production_decrease.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate
84,2025Q1,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.587,0.160,0.253
74,2025Q1,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.693,0.143,0.164
32,2025Q4,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.649,0.138,0.213
33,2025Q4,Madencilik ve Taş Ocakçılığı,0.602,0.130,0.268
50,2025Q3,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.620,0.129,0.252
13,2026Q1,Toptan ve Perakende Ticaret,0.699,0.128,0.173
76,2025Q1,Toptan ve Perakende Ticaret,0.684,0.124,0.192
83,2025Q1,"Kültür, Sanat Eğlence, Dinlence ve Spor",0.640,0.123,0.236
24,2025Q4,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.720,0.123,0.157
43,2025Q3,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.723,0.117,0.160


## Analysis 4: Sectors with the Highest Uncertainty

This analysis identifies the sectors with the highest no-opinion rate.  
A higher no-opinion rate may indicate uncertainty about future service/production expectations.

In [17]:
top_service_production_uncertainty = df_service_production[
    ["period", "economic_activity", "maintain_or_increase_rate", "decrease_rate", "no_opinion_rate"]
].sort_values(by="no_opinion_rate", ascending=False)

top_service_production_uncertainty.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate
33,2025Q4,Madencilik ve Taş Ocakçılığı,0.602,0.130,0.268
66,2025Q2,İnşaat,0.679,0.056,0.265
67,2025Q2,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.624,0.116,0.259
84,2025Q1,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.587,0.160,0.253
50,2025Q3,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.620,0.129,0.252
49,2025Q3,İmalat,0.636,0.113,0.251
15,2026Q1,Madencilik ve Taş Ocakçılığı,0.688,0.067,0.244
81,2025Q1,Gayrimenkul Faaliyetleri,0.672,0.087,0.241
48,2025Q3,Madencilik ve Taş Ocakçılığı,0.678,0.085,0.237
83,2025Q1,"Kültür, Sanat Eğlence, Dinlence ve Spor",0.640,0.123,0.236


## Analysis 5: Service/Production Outlook Score

This analysis creates a simple service/production outlook score by subtracting the decrease expectation rate from the maintain/increase expectation rate.  
A higher score indicates a more positive service/production expectation.

In [18]:
df_service_production["service_production_outlook_score"] = (
    df_service_production["maintain_or_increase_rate"] - df_service_production["decrease_rate"]
)

service_production_outlook = df_service_production[
    ["period", "economic_activity", "maintain_or_increase_rate", "decrease_rate", "no_opinion_rate", "service_production_outlook_score"]
].sort_values(by="service_production_outlook_score", ascending=False)

service_production_outlook.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate,service_production_outlook_score
34,2025Q3,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.806,0.041,0.153,0.765
35,2025Q3,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.801,0.039,0.159,0.762
68,2025Q1,"Elektrik, Gaz, Buhar ve İklimlendirme Üretimi ...",0.797,0.037,0.167,0.760
17,2025Q4,Bilgi ve İletişim,0.802,0.042,0.156,0.760
19,2025Q4,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.786,0.038,0.175,0.748
51,2025Q2,Bilgi ve İletişim,0.790,0.047,0.163,0.743
18,2025Q4,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.792,0.050,0.157,0.742
69,2025Q1,Finans ve Sigorta Faaliyetleri,0.787,0.051,0.162,0.736
71,2025Q1,Bilgi ve İletişim,0.770,0.048,0.182,0.722
52,2025Q2,Diğer Hizmet Faaliyetleri,0.775,0.053,0.172,0.722


## Key Insights

- Service/production expectations are generally positive across all periods, with average maintain/increase rates around 70%–73%.
- The average maintain/increase expectation gradually increases from 2025 Q1 to 2026 Q1.
- Professional, Scientific and Technical Activities, Information and Communication, and Human Health and Social Work Activities appear among the strongest sectors in terms of service/production expectations.
- Water Supply, Sewerage and Waste Management, Accommodation and Food Service Activities, and Mining and Quarrying show relatively higher decrease expectations in some periods.
- Mining and Quarrying, Construction, and Water Supply-related activities show higher no-opinion rates, indicating stronger uncertainty.
- The service/production outlook score helps compare sectors by combining positive and negative expectations into a single metric.